plugins/modules/ecs_cluster.py

```diff
@@ -178,11 +257,36 @@ def main():
     results = dict(changed=False)
     if module.params['state'] == 'present':
         if existing and 'status' in existing and existing['status'] == "ACTIVE":
-            results['cluster'] = existing
+            # Pull requested and existing capacity providers and strategies.
+            requested_cp = module.params['capacity_providers']
+            requested_cps = module.params['capacity_provider_strategy']
+            existing_cp = existing['capacityProviders']
+            existing_cps = existing['defaultCapacityProviderStrategy']
+            if requested_cp is None:
+                requested_cp = []
```

Comment:
```suggestion
                requested_cp = []
```

This breaks backwards compatibility.  
Until now, `ecs_cluster` can only create clusters and the users must add the capacty providers and strategies by their own (_aws cli, custom boto3 script, click in the web console ..._).  
Rerun the `ecs_cluster` task won't change this.  

But with this change, all manually added capacity providers and strategies will be dropped - accidentially. (_if the users doesn't read the changelogs and add the capacity providers and strategies_)  
So for the initial release, we must add a parameter `purge_capacity_providers: default(false)`.

What do you think @alinabuzachis @tremble

Reply:
>  So for the initial release, we must add a parameter purge_capacity_providers: default(false).

for the record ...  
we can deprecate and drop this 1-2 major releases later ...